### PR TITLE
Add documentation provider

### PR DESCRIPTION
**Pull request by vkatsuba**
For case when last version(`7.0.1`) of project plugin `rebar3_hex` is used for publication of documentation to `hex.pm`, can be received second warning:
```sh
$ rebar3 hex publish docs
...
===> Verifying dependencies...
===> docs argument given, will not publish package
===> No documentation provider was configured, docs will not be generated
```
For this case was added provider as suggested in `rebar3_hex` documentation.

### CODE REVIEW

File: rebar.config [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 {project_plugins, [rebar3_proper,
                    rebar3_hex,
-                   {rebar3_edoc_extensions, "1.0.0"}]}.
+                   rebar3_edoc_extensions]}.
```
Comment (dumbbell) [inline rebar.config:9]: Could you please remove this unrelated change?

**Review by dumbbell — CHANGES_REQUESTED** [review]
Could you please add the explanation you wrote in the pull request to the commit message too? This way the commit history is self-explanatory; no need to go to GitHub.

**Comment by vkatsuba** [comment]
> Could you please add the explanation you wrote in the pull request to the commit message too? This way the commit history is self-explanatory; no need to go to GitHub.

Done.

File: rebar.config (hunk 1 above)
Comment (vkatsuba) [reply rebar.config]: Done. Do I need move those changes in separate commit or open new PR for it? Or for now this is not necessary at all?

**Review by dumbbell — APPROVED** [review]

**Comment by dumbbell** [comment]
Thank you!
